### PR TITLE
Hoh fix teardown interfaces

### LINKED ISSUE #129: Network interfaces may accumulate on host

### PR DESCRIPTION
**Pull request by hoh**
Fixes #129

### CODE REVIEW

File: firecracker/microvm.py [hunk 1]
```diff
@@ -369,23 +369,31 @@ async def unix_client_connected(*_):
...
+            await writer.drain()
 
-        ack: bytes = await reader.readline()
-        logger.debug(f"ack={ack.decode()}")
+            ack: bytes = await reader.readline()
+            logger.debug(f"ack={ack.decode()}")
 
-        msg: bytes = await reader.readline()
-        logger.debug(f"msg={msg}")
+            msg: bytes = await reader.readline()
```
Comment (odesenfans) [inline firecracker/microvm.py:387]: readline + binary output? Seems weird, and you don't use decode in the other logs so I imagine it's actually a string and the type hint is wrong.

File: firecracker/microvm.py [hunk 2]
```diff
@@ -369,23 +369,31 @@ async def unix_client_connected(*_):
...
 
-        msg2: bytes = await reader.readline()
-        logger.debug(f"msg2={msg2}")
+            msg2: bytes = await reader.readline()
+            logger.debug(f"msg2={msg2}")
 
-        if msg2 != b"STOPZ\n":
-            logger.error("Unexpected response from VM")
+            if msg2 != b"STOPZ\n":
+                logger.error("Unexpected response from VM")
```
Comment (odesenfans) [inline firecracker/microvm.py:394]: Maybe log the response?

**Review by odesenfans — COMMENTED** [review]
Some minor comments.

File: firecracker/microvm.py (hunk 1 above)
Comment (hoh) [reply firecracker/microvm.py:387]: The type hint is correct: readline can be used on bytes arrays.

File: firecracker/microvm.py [hunk 3]
```diff
@@ -369,23 +369,31 @@ async def unix_client_connected(*_):
 
     async def shutdown(self):
         logger.debug(f"Shutown vm={self.vm_id}")
```
[the remark this answers is not in the document]
Comment (hoh) [reply firecracker/microvm.py:371]: The issue is not created by this PR. Moved into new issue.

**Pull request merged**